### PR TITLE
tighten up repo docs for contributor focus

### PR DESCRIPTION
**Pull request by incanus**
The goal here is that this repo targets folks who are hacking on GL directly — **not** merely installing a product such as Mapbox Mobile. There will be mapbox.com changes to support this, but this PR takes care of this repo as a landing place for this audience.

### CODE REVIEW

**Comment by incanus** [comment]
Other tasks: 

- [ ] Deprecate/clean up https://github.com/mapbox/mapbox-gl-native/wiki/Installing-Mapbox-GL-for-iOS

**Comment by incanus** [comment]
>  Deprecate/clean up https://github.com/mapbox/mapbox-gl-native/wiki/Installing-Mapbox-GL-for-iOS

Per chat with @1ec5, we are going to reduce this to merely a resource for IB integration, as well as ticket a longer-term solution for that documentation.

**Comment by incanus** [comment]
IB consolidation: https://github.com/mapbox/mapbox-gl-native/wiki/Using-Mapbox-Mobile-in-iOS-Interface-Builder

Future work not necessarily part of this milestone: https://github.com/mapbox/guides/issues/418

File: INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+![setting access token in Xcode scheme](https://cloud.githubusercontent.com/assets/162976/5349358/0a086f00-7f8c-11e4-8433-bdbaccda2b58.png)
+
+For Linux, set the environment variable `MAPBOX_ACCESS_TOKEN` to your token.
+
+For Android, gradle will take the value of `MAPBOX_ACCESS_TOKEN` and save it to `android/java/MapboxGLAndroidSDKTestApp/src/main/res/raw/token.txt` where the app will read it from.
+
+# Style
+
+Some styles in JSON format are included at `./styles`. See the [style spec](https://github.com/mapbox/mapbox-gl-style-spec) for more details.
```
Comment (1ec5) [inline INSTALL.md]: This paragraph should point to https://github.com/mapbox/mapbox-gl-styles for contributing to the bundled styles.

**Comment by incanus** [comment]
Will need to his these when the URL slug reorg happens. 

```
$ ag gl-ios
ios/docs/pod-README.md
5:For more information, check out [our online overview](https://www.mapbox.com/mapbox-gl-ios/). 

ios/Mapbox-iOS-SDK.podspec
8:  m.homepage         = 'https://www.mapbox.com/mapbox-gl-ios/'
13:  m.documentation_url = 'https://www.mapbox.com/mapbox-gl-ios'
16:    :http => "http://mapbox.s3.amazonaws.com/mapbox-gl-native/ios/builds/mapbox-gl-ios-#{m.version.to_s}.zip",

platform/ios/MGLMapView.mm
45:NSString *const MGLMapboxSetupDocumentationURLDisplayString = @"mapbox.com/guides/first-steps-gl-ios";
```

**Comment by mb12** [comment]
@incanus Is it possible to document steps to build external dependencies (libuv, curl, sqlite, libzip etc)?

**Comment by incanus** [comment]
I think this probably belongs in the domain of http://github.com/mapbox/mason, but I'll defer to @springmeyer.

**Pull request merged**